api/user.go

```diff
@@ -354,14 +380,22 @@ func (a *UserAPI) UpdateUserByID(ctx *gin.Context) {
 	withID(ctx, "id", func(id uint) {
 		var user *model.UserExternalWithPass
 		if err := ctx.Bind(&user); err == nil {
-			if oldUser := a.DB.GetUserByID(id); oldUser != nil {
-				if !user.Admin && oldUser.Admin && a.DB.CountUser(&model.User{Admin: true}) == 1 {
+			oldUser, err := a.DB.GetUserByID(id)
+			if success := successOrAbort(ctx, 500, err); !success {
+				return
+			}
+			if oldUser != nil {
+				adminCount, err := a.DB.CountUser(&model.User{Admin: true})
+				if success := successOrAbort(ctx, 500, err); !success {
+					return
+				}
+				if !user.Admin && oldUser.Admin && adminCount == 1 {
 					ctx.AbortWithError(400, errors.New("cannot delete last admin"))
 					return
 				}
 				internal := a.toInternalUser(user, oldUser.Pass)
 				internal.ID = id
-				a.DB.UpdateUser(internal)
+				successOrAbort(ctx, 500, a.DB.UpdateUser(internal))
```

Comment:
return if error

Reply:
The current branch head is already the correct one I think? Maybe you are not on the tip of the branch?